File: src/main/java/com/graqr/threshr/model/redsky/product/Nutrient.java
package com.graqr.threshr.model.redsky.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record Nutrient(String name,
                       Double quantity,
                       @JsonProperty("unit_of_measurement")
                       String unitOfMeasurement,
                       Double percentage) {
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/product/Nutrient.kt
package com.graqr.threshr.circleStore.model.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class Nutrient(
    val name: String,
    val quantity: Double,
    @param:JsonProperty("unit_of_measurement") val unitOfMeasurement: String,
    val percentage: Double
) {}
